libs/application/template-api-modules/src/lib/modules/templates/general-petition/general-petition.service.ts

```diff
@@ -0,0 +1,113 @@
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { Inject, Injectable } from '@nestjs/common'
+import { TemplateApiModuleActionProps } from '../../../types'
+import { SharedTemplateApiService } from '../../shared'
+import {
+  EndorsementListApi,
+  EndorsementMetadataDtoFieldEnum,
+} from './gen/fetch/endorsements'
+
+const CREATE_ENDORSEMENT_LIST_QUERY = `
+  mutation EndorsementSystemCreatePartyLetterEndorsementList($input: CreateEndorsementListDto!) {
+    endorsementSystemCreateEndorsementList(input: $input) {
+      id
+    }
+  }
+`
+
+/**
+ * We proxy the auth header to the subsystem where it is resolved.
+ */
+interface FetchParams {
+  url: string
+  init: RequestInit
+}
+
+interface RequestContext {
+  init: RequestInit
+}
+
+interface Middleware {
+  pre?(context: RequestContext): Promise<FetchParams | void>
+}
+class ForwardAuthHeaderMiddleware implements Middleware {
+  constructor(private bearerToken: string) {}
+
+  async pre(context: RequestContext) {
+    context.init.headers = Object.assign({}, context.init.headers, {
+      authorization: this.bearerToken,
+    })
+  }
+}
+
+type ErrorResponse = {
+  errors: {
+    message: string
+  }
+}
+type EndorsementListResponse =
+  | {
+      data: {
+        endorsementSystemCreateEndorsementList: {
+          id: string
+        }
+      }
+    }
+  | ErrorResponse
+
+@Injectable()
+export class GeneralPetitionService {
+  constructor(
+    private endorsementListApi: EndorsementListApi,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private readonly sharedTemplateAPIService: SharedTemplateApiService,
+  ) {}
+
+  endorsementListApiWithAuth(token: string) {
+    return this.endorsementListApi.withMiddleware(
+      new ForwardAuthHeaderMiddleware(token),
+    )
+  }
+
+  async createEndorsementList({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps) {
+    const endorsementList: EndorsementListResponse = await this.sharedTemplateAPIService
+      .makeGraphqlQuery(auth.authorization, CREATE_ENDORSEMENT_LIST_QUERY, {
+        input: {
+          title: application.answers.listName,
+          description: application.answers.aboutList,
+          endorsementMetadata: [
+            { field: EndorsementMetadataDtoFieldEnum.address },
+          ],
+          tags: ['partyLetter2021'],
```

Comment:
Is the tag `partyLetter2021` not reserved for the party letter registration in the endorsement system?